api/storage_proxy.cc

```diff
@@ -201,29 +202,38 @@ void set_storage_proxy(http_context& ctx, routes& r) {
     });
 
     sp::get_hinted_handoff_enabled.set(r, [&ctx](std::unique_ptr<request> req)  {
-        auto enabled = ctx.db.local().get_config().hinted_handoff_enabled();
-        return make_ready_future<json::json_return_type>(enabled.to_configuration_string());
+        auto filter = service::get_storage_proxy().local().get_hints_host_filter();
+        return make_ready_future<json::json_return_type>(!filter.is_disabled_for_all());
     });
 
     sp::set_hinted_handoff_enabled.set(r, [](std::unique_ptr<request> req)  {
-        //TBD
-        unimplemented();
         auto enable = req->get_query_param("enable");
-        return make_ready_future<json::json_return_type>(json_void());
+        auto filter = (enable == "true")
+                ? db::hints::host_filter(db::hints::host_filter::enabled_for_all_tag {})
+                : db::hints::host_filter(db::hints::host_filter::disabled_for_all_tag {});
+        return service::get_storage_proxy().invoke_on_all([filter] (service::storage_proxy& sp) {
+            sp.set_hints_host_filter(filter);
```

Comment:
Changing a configuration is an asynchronous operation therefore `sp.set_hints_host_filter(filter)` should return a future which should be what the corresponding manager::on_host_filter_update(...)` returns (as I described before).

The api/storage_proxy.cc code implies that - exactly as it should be.